pkg/cloud/openstack/machine/actuator.go

```diff
@@ -94,6 +104,17 @@ func (oc *OpenstackClient) Create(ctx context.Context, cluster *clusterv1.Cluste
 		return nil
 	}
 
+	// create floatingip if needed
+	if providerSpec.FloatingIP == "" && machineSpec.ExternalNetworkID != "" {
+		fip, err := networkService.CreateFloatingIP(machineSpec.ExternalNetworkID)
```

Comment:
Before creating the FIP, try to find an existing one with the description `machine.Meta.Uid` (?)